### PR TITLE
Remove boto as a dependency

### PR DESCRIPTION
**Pull request by isobit**
Fixes:
- https://github.com/RaRe-Technologies/smart_open/issues/391
- https://github.com/RaRe-Technologies/smart_open/issues/409

Related to:
- https://github.com/RaRe-Technologies/smart_open/pull/413
- https://github.com/RaRe-Technologies/smart_open/issues/468
- https://github.com/RaRe-Technologies/smart_open/pull/408

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -41,6 +41,7 @@ def read(fname):
     'moto[server]',
     'pathlib2',
     'responses',
+    'boto',
```
Comment (mpenkov) [inline setup.py:44]: Why do our tests still need boto?

Comment (isobit) [reply setup.py:44]: There's a `test_old` method which uses an old `boto` bucket: https://github.com/RaRe-Technologies/smart_open/blob/ecce29ebb3b0dee519f6939e924ee4bb5974ca33/smart_open/tests/test_s3.py#L576

I could at least leave a comment here with an explanation. I suppose alternatively the support for passing in old boto buckets could be deprecated and the test removed.

Comment (mpenkov) [reply setup.py:44]: Yeah, I think it's best to remove it altogether.

Sorry for the late response.

File: README.rst [hunk 2]
```diff
@@ -29,9 +29,9 @@ What?
 Why?
 ====
 
-Working with large remote files, for example using Amazon's `boto <http://docs.pythonboto.org/en/latest/>`_ and `boto3 <https://boto3.readthedocs.io/en/latest/>`_ Python library, is a pain.
-``boto``'s ``key.set_contents_from_string()`` and ``key.get_contents_as_string()`` methods only work for small files, because they're loaded fully into RAM, no streaming.
-There are nasty hidden gotchas when using ``boto``'s multipart upload functionality that is needed for large files, and a lot of boilerplate.
+Working with large remote files, for example using Amazon's `boto3 <https://boto3.readthedocs.io/en/latest/>`_ Python library, is a pain.
+``boto3``'s ``key.set_contents_from_string()`` and ``key.get_contents_as_string()`` methods only work for small files, because they're loaded fully into RAM, no streaming.
```
Comment (piskvorky) [inline README.rst:33]: Does boto3 even have `set_contents_from_string` / `get_contents_as_string `?

Comment (isobit) [reply README.rst:33]: Ah good point, I don't think it does. `boto3` has `upload_fileobj()` and `download_fileobj()` which do handle multipart streaming. The point about having a unified pythonic API still stands though obviously.

**Comment by mpenkov** [comment]
Thank you for your contribution and your patience @isobit ! And congrats on your first PR to smart_open 🥇

**Pull request merged**